### PR TITLE
Fix finding of deep links

### PR DESCRIPTION
**Pull request by StephenWeatherford**
In the particular case I looked at we were picking up the wrong version of a module in determining dependencies, thus ending up with missing modules copied to node_modules in cosmos. Now look at subdependencies when listed, and not just "requires".

https://github.com/Microsoft/vscode-cosmosdb/issues/1020

### CODE REVIEW

File: dev/package-lock.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
             "integrity": "sha512-PDkSRY7KltW3M60hSBlerxI8SFPXsO3AL/aRVsO4Kh9IHRW74Ih75gUuTd/aE4LSSFqypb10UIX3QzOJwBQMGQ==",
             "dev": true,
             "requires": {
-                "@types/events": "*",
-                "@types/node": "*"
+                "@types/events": "1.2.0",
```
Comment (EricJizbaMSFT) [inline dev/package-lock.json]: Did you update this package lock before or after realizing you were on npm 5.6?

**Comment by StephenWeatherford** [comment]
> Have you every looked at `npm ls --json`? It seems like a much easier tree to parse. Here's what I get for "engine.io". I may be wrong, but I think you just follow "dependencies" all the way down to get the whole list.

I did. Didn't look like it provided any better info, although I could be wrong.

File: dev/package-lock.json (hunk 1 above)
Comment (StephenWeatherford) [reply dev/package-lock.json]: Had thought it was after but obviously not.  Fixed.

Comment (StephenWeatherford) [reply dev/package-lock.json]: I think I did an npm update as well, hence still a lot of changes but not the same kind.

**Comment by StephenWeatherford** [comment]
> > Have you every looked at `npm ls --json`? It seems like a much easier tree to parse. Here's what I get for "engine.io". I may be wrong, but I think you just follow "dependencies" all the way down to get the whole list.
> 
> I did. Didn't look like it provided any better info, although I could be wrong.

Looks like it might warrant another look, but I'm not worry about it right now.

**Pull request merged**